### PR TITLE
Add Management Cluster Promxy Config

### PR DESCRIPTION
**Pull request by QuentinBisson**
This should be replacing the pmo resource

### CODE REVIEW

**Comment by QuentinBisson** [comment]
Tested on ginger:
`time="2021-04-01T13:31:27Z" level=debug msg="Updating targets from discovery manager"
time="2021-04-01T13:31:27Z" level=debug msg="Updating targets from discovery manager: [100.64.165.167:9090 100.64.165.167:8080 100.64.43.1:9090 100.64.43.1:8080 100.64.109.252:9090 100.64.109.252:8080]"
10.0.5.43 - - [01/Apr/2021 13:31:30] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000064 
10.0.5.43 - - [01/Apr/2021 13:31:31] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000055 
time="2021-04-01T13:31:32Z" level=debug msg="Updating targets from discovery manager"
time="2021-04-01T13:31:32Z" level=debug msg="Updating targets from discovery manager: [100.64.109.252:9090 100.64.109.252:8080 100.64.43.1:9090 100.64.43.1:8080]"
10.0.5.43 - - [01/Apr/2021 13:31:35] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000049 
10.0.5.43 - - [01/Apr/2021 13:31:36] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000056 
time="2021-04-01T13:31:37Z" level=debug msg="Updating targets from discovery manager"
time="2021-04-01T13:31:37Z" level=debug msg="Updating targets from discovery manager: [100.64.139.207:9090 100.64.139.207:8080 100.64.43.1:9090 100.64.43.1:8080 100.64.109.252:9090 100.64.109.252:8080]"
10.0.5.43 - - [01/Apr/2021 13:31:40] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000051 
10.0.5.43 - - [01/Apr/2021 13:31:41] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000052 
10.0.5.43 - - [01/Apr/2021 13:31:45] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000049 
10.0.5.43 - - [01/Apr/2021 13:31:46] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000054 
10.0.5.43 - - [01/Apr/2021 13:31:50] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000055 
10.0.5.43 - - [01/Apr/2021 13:31:51] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000053 
10.0.5.43 - - [01/Apr/2021 13:31:55] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000065 
10.0.5.43 - - [01/Apr/2021 13:31:56] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000062 
10.0.5.43 - - [01/Apr/2021 13:32:00] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000065 
10.0.5.43 - - [01/Apr/2021 13:32:01] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000055 
10.0.5.43 - - [01/Apr/2021 13:32:05] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000052 
10.0.5.43 - - [01/Apr/2021 13:32:06] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000057 
10.0.5.43 - - [01/Apr/2021 13:32:10] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000059 
10.0.5.43 - - [01/Apr/2021 13:32:11] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000053 
10.0.5.43 - - [01/Apr/2021 13:32:15] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000050 
10.0.5.43 - - [01/Apr/2021 13:32:16] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000060 
10.0.5.43 - - [01/Apr/2021 13:32:20] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000058 
10.0.5.43 - - [01/Apr/2021 13:32:21] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000055 
10.0.5.43 - - [01/Apr/2021 13:32:25] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000055 
10.0.5.43 - - [01/Apr/2021 13:32:26] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000054 
10.0.5.43 - - [01/Apr/2021 13:32:30] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000058 
10.0.5.43 - - [01/Apr/2021 13:32:31] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000053 
10.0.5.43 - - [01/Apr/2021 13:32:35] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000054 
10.0.5.43 - - [01/Apr/2021 13:32:36] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000053 
10.0.5.43 - - [01/Apr/2021 13:32:40] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000053 
10.0.5.43 - - [01/Apr/2021 13:32:41] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000051 
10.0.5.43 - - [01/Apr/2021 13:32:45] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000062 
10.0.5.43 - - [01/Apr/2021 13:32:46] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000056 
10.0.5.43 - - [01/Apr/2021 13:32:50] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000051 
10.0.5.43 - - [01/Apr/2021 13:32:51] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000060 
10.0.5.43 - - [01/Apr/2021 13:32:55] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000064 
10.0.5.43 - - [01/Apr/2021 13:32:56] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000056 
10.0.5.43 - - [01/Apr/2021 13:33:00] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000057 
10.0.5.43 - - [01/Apr/2021 13:33:01] "GET /promxy/-/healthy HTTP/1.1 200 23" 0.000065 
10.0.5.43 - - [01/Apr/2021 13:33:05] "GET /promxy/-/ready HTTP/1.1 200 21" 0.000050 
`

**Review by TheoBrigitte — APPROVED** [review]
Looks about right

**Comment by mig4** [comment]
What does this do?

**Comment by QuentinBisson** [comment]
Instead of having one server group per prometheus, we have only one that can match all the pmo managed prometheus in the promxy app

File: config/promxy.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          target_label: __path_prefix__
+          replacement: /$1
+          action: replace
+        kubernetes_sd_configs:
+        - api_server: https://kubernetes.default:443
+          role: pod
+          bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+          tls_config:
+            ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+            insecure_skip_verify: true
```
Comment (mig4) [inline config/promxy.yaml:35]: It doesn't work without this? 😬

File: config/promxy.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        timeout: 2m0s
+        ignore_error: true
```
Comment (mig4) [inline config/promxy.yaml:37]: What's this do?

File: config/promxy.yaml (hunk 1 above)
Comment (QuentinBisson) [reply config/promxy.yaml:35]: We have this in pmo so I did not change it here. We can test it later for sure as we have a story to clean this up

File: config/promxy.yaml (hunk 2 above)
Comment (QuentinBisson) [reply config/promxy.yaml:37]: This ignore errors in conjunction with the timeout so if a prometheus is slow to respond, i.e. more than 2 minutes, promxy will not return an error but the result of the other prometheus it gets data from

**Review by JosephSalisbury — APPROVED** [review]

**Pull request merged**